workflow/sync/semaphore.go

```diff
@@ -3,69 +3,63 @@ package sync
 import (
 	"fmt"
 	"strings"
-	"sync"
 	"time"
 
 	log "github.com/sirupsen/logrus"
 	sema "golang.org/x/sync/semaphore"
 )
 
-type PrioritySemaphore struct {
+type prioritySemaphore struct {
```

Comment:
Not relevant on this PR but just a note: 
We should get rid of the semaphore interface and use this directly.

There is no need for indirection anymore given we just have a semaphore now.